records/2024-04-11.2FA-magic-link.md

```diff
@@ -0,0 +1,135 @@
+# Add magic links to 2FA emails
+
+Date: 2024-04-11
+
+## Status
+
+**DRAFT**.
+
+_VALUES: (DRAFT, IN REVIEW, APPROVED, REJECTED)_
+
+## Context
+
+Notify requires users to manually copy two factor authentication codes from their email then paste it into a text box on the site in order to log into Notify. This process is cumbersome and not that accessible, or easy to use.
+
+The aim of this initiative is to improve the accessibility and ease of use of Notify's login process.
+
+## Options
+
+### Increase the visibility of 2FA codes in emails
+The simplest approach is to increase the visiblility of codes in the email to make the code more obvious and easier to interact with:
+- Place the code on it's own line in the email
+- Bold or otherwise highlight the text
+- Increase font size
+
+While increasing visibility is an improvement, it does not address the cumbersome amount of actions needed to complete the login:
+
+```mermaid
+flowchart LR
+    subgraph Email Inbox Tab
+        id1(Open email) --> id2(Find the code)--> id3(Highlight it)--> id4(Copy it)
+    end
+	subgraph Notify Tab
+        id0(Enter username and password)
+        id6(Select 2FA code text box)-->id7(Paste in the code)-->id8(Click Submit)
+    end
+    id0 --Tab Switch--> id1
+    id4 --Tab Switch--> id6
+    id8-->id9((X))
+
+```
+
+### Use magic links
+By using a magic link that the user can simply click on to complete the login process, we reduce the number of manual interactions required by the user to log in, thus making the process less obstructive and feel more natural.
```

Comment:
Can we please explain that what we define here as magic links might not be true magic links as generally understood in the industry? i.e. we still require a password prior to sending the link, whereas a magic link solution would only require the email information, if I understood correctly.

Reply:
Your understanding is correct, I've added detail on how our implementation would differ from the typical industry standard.